补充

xiangyuli-bio opened the pull request:
1.data.py:把TestDataModule类（用于加载缓存的三维Chembert的类）为了方便理解
  改名成TDCDataModule_Local类 所有的类都基于TDCDataModule编写
  共使用：TDCDataModule类（基类），CSVDataModule类（用于直接读取CSV的类），TDCDataModule_Local(本地加载三维Chembert类），TDCDataModule_Double(用于同时加载Morgan和Chemberta的特征类）
2.baseline_dataModule.py
  数据加载使用TDCDataModule类,用于训练base实验，单独的morgan或者chember实验
3.morgan_chembert_datamodule.py 数据加载使用TDCDataModule_Double类
  用于训练morgan和chembert同时加载的实验
4.mrogan_chembert_model.py 配合第3项
5.morgan_model.py 配合第2项
6.utils.py 修改可返回drugfeature列表

xiangyuli-bio committed:
补充

1.data.py:把TestDataModule类（用于加载缓存的三维Chembert的类）为了方便理解
  改名成TDCDataModule_Local类 所有的类都基于TDCDataModule编写
  共使用：TDCDataModule类（基类），CSVDataModule类（用于直接读取CSV的类），TDCDataModule_Local(本地加载三维Chembert类），TDCDataModule_Double(用于同时加载Morgan和Chemberta的特征类）
2.baseline_dataModule.py
  数据加载使用TDCDataModule类,用于训练base实验，单独的morgan或者chember实验
3.morgan_chembert_datamodule.py 数据加载使用TDCDataModule_Double类
  用于训练morgan和chembert同时加载的实验
4.mrogan_chembert_model.py 配合第3项
5.morgan_model.py 配合第2项
6.utils.py 修改可返回drugfeature列表

fireinice commented on src/models/morgan_model.py:
```diff
@@ -0,0 +1,283 @@
+from typing import Optional
+
+
+class MLP(nn.Module):
```
这些相同的代码不要在不同的文件里重复定义

fireinice commented on src/models/morgan_chembert_model.py:
```diff
@@ -0,0 +1,305 @@
+        return embeddings
+
+# 二维的Chembert或者Morgan作为cls拼接三维的Protbert
+class DrugProteinAttention(nn.Module):
```
从之前的类中继承，相同的函数不要不断复制

fireinice reviewed (commented):
不要把代码不断复制，从一个实现的类出发，其它的从文件中import进来，或者继承过来

xiangyuli-bio committed:
第二次完善

1.baseline_datamodule.py 完善per_tok的判断
2.morgan_chembert_datamodule.py 模型代码移出
3.achitectures.py 基类：DrugProteinAttention
  同时加载mrogan和chembert特征类DrugProteinAttention_Double
4.data.py  完善Bug
5.base_model.py 加入距离惩罚损失函数
6.morgan_chembert_model.py和morgan_model.py self.log只能记录标量
  需要加入判断
7.lightning_model.py 统一分类验证指标
8.补充：base_model.py 基类 morgan_model.py
  用于单独的morgan或者chembert实验 morgan_chembert_model.py
  用于同时加载morgan和chembert实验
